needle.pm

```diff
@@ -80,7 +80,7 @@ sub new {
         catch {
             warn "broken json $jsonfile: $_";
         };
-        return undef unless $json;
+        return undef unless $json && $json eq 'HASHREF';
```

Comment:
Where does `HASHREF` come from? This looks similar to the usual `ref $json eq 'HASH'` check. Did you meant that?

Reply:
yes, I did. But even if I change it now tests fail and I remember at the time of writing tests could not pass. I will remove that commit.